src/components/IconLibrary/IconLibrary.js

```diff
@@ -212,32 +244,63 @@ function createIconSections(icons, filteredIcons) {
       return groups[size].length !== 0;
     })
     .map(size => (
-      <section key={size} className="icon-size">
+      <section key={size} className="icon-size" aria-labelledby={`icon-h2`}>
         <header>
-          <h2 className="icon-h2">{isNaN(size) ? size : `${size}x${size}`}</h2>
+          <h2 className={`icon-h2`}>
+            {isNaN(size) ? size : `${size}x${size}`}
+          </h2>
         </header>
-        <div className="icon-container">
-          {groups[size]
-            .filter(icon => filteredIcons.indexOf(icon.name) !== -1)
-            .map(renderIcon)}
-        </div>
+        <ul className="icons-list">
+          {renderIconList(groups[size], filteredIcons)}
+        </ul>
       </section>
     ));
 }
 
+/**
+ * render list of available icons for this category.
+ * if none are available, return a blank block
+ */
+
+function renderIconList(categoryArray, filteredList) {
+  const displayedIconsList = categoryArray.filter(
+    icon => filteredList.indexOf(icon.name) !== -1
+  );
+
+  const displayedIcons = displayedIconsList.map(renderIcon);
+
+  if (displayedIcons.length > 0) {
+    return displayedIcons;
+  }
+
+  return (
+    <li className="icon__container">
+      <div className="icon__aspect-ratio">
+        <div className="icon__card" />
+        <h5 className="icon__card-title" title="No results in this size.">
+          No results in this size.
+        </h5>
+      </div>
+    </li>
+  );
+}
+
 /**
  * Renders an individual icon
+ * passing `null` renders no-results icon block
  */
 function renderIcon(icon) {
   return (
-    <div key={icon.name} className="icon">
-      <div className="icon__card">
-        <icon.Component />
+    <li key={icon.name} className="icon__container">
+      <div className="icon__aspect-ratio">
```

Comment:
We could use the built in aspect ratio class names here if you want? https://carbon-elements.netlify.com/grid/examples/preview/#aspect-ratio

Reply:
oh yeah i should've used those instead of re-writing 😅